halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/versions/BomVersionCommand.java

```diff
@@ -97,4 +105,9 @@ protected void executeThis() {
 
     AnsiPrinter.out.println(result);
   }
+
+  private static boolean isProjectVersionFormat(String version) {
+    // Matches #, #.#, #.#.#, ...
+    return version.matches("\\d+(\\.\\d+)*");
```

Comment:
I would delegate to https://github.com/spinnaker/halyard/blob/master/halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/registry/v1/Versions.java#L54 to check if a version is a branch or local, and assume all other versions can at least be looked up.